logic/strategy/src/one_hop_connection.rs

```diff
@@ -0,0 +1,358 @@
+//! ## One hop connection strategy
+//! This strategy is designed to maintain 1-hop pathes to certain destination peers,
+//! where there is at least one relay node that is not part of the destination peer set.
+//!
+//! There should be outgoing channels to the destination peers.
+use async_trait::async_trait;
+use chain_actions::channels::ChannelActions;
+use futures::StreamExt;
+use hopr_db_sql::api::peers::PeerSelector;
+use hopr_db_sql::HoprDbAllOperations;
+use hopr_internal_types::channels::{ChannelDirection, ChannelEntry};
+use hopr_primitive_types::prelude::*;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::collections::HashMap;
+use std::fmt::{Debug, Display, Formatter};
+use std::ops::{Add, Sub};
+use std::str::FromStr;
+use tracing::{debug, error, info};
+use validator::Validate;
+
+use crate::errors::{self, Result};
+use crate::strategy::SingularStrategy;
+use crate::types::ChannelDecision;
+use crate::Strategy;
+
+#[cfg(all(feature = "prometheus", not(test)))]
+use hopr_metrics::metrics::SimpleCounter;
+
+#[cfg(all(feature = "prometheus", not(test)))]
+lazy_static::lazy_static! {
+    static ref METRIC_COUNT_OPENS: SimpleCounter =
+        SimpleCounter::new("hopr_strategy_one_hop_connection_opened_channels_count", "Count of open channel decisions").unwrap();
+    static ref METRIC_COUNT_CLOSURES: SimpleCounter =
+        SimpleCounter::new("hopr_strategy_one_hop_connection_closed_channels_count", "Count of close channel decisions").unwrap();
```

Comment:
The first major issue of this strategy will be that it is cost unaware. Each of those operations incurs a physical cost that cannot be accounted for in the current code, so the strategy is not taking it into considerations, making it potentially wasteful.

Reply:
We should improve that by adding a general rate-limiter of open/close TXs.

Btw, you probably meant "financial cost", rather than "physical" - I cannot see how this strategy can potentially melt your Dappnode :)